### ISSUE TITLE
JWT + Load Balancer Not Working

### ISSUE THREAD

**Issue by JKing-B16** [nhost/hasura-backend-plus#457]
We just tried to release a project where we were running two VM's each with Hasura and hasura-backend-plus. We are behind a Google load balancer. So a user randomly makes requests to one of the VM's. Hasura worked fine for non-logged in users, but the JWT is getting rejected if you aren't hitting the same server that issues it. Since both servers talk to the same database, could this be an issue with Hasura or Hasura-Backend-Plus caching the JWT and doesn't bother to double check the DB? Is there a ready made solution to this?

hasura/graphql-engine:v1.3.2
nhost/hasura-backend-plus:latest

**Comment by elitan** [comment]
Do the two instances share the exact same JWT algorithm and JWT key?

Check the following variables:

**Hasura**
`HASURA_GRAPHQL_JWT_SECRET`

**Hasura Backend Plus**
`JWT_ALGORITHM` and `JWT_KEY`

**Comment by JKing-B16** [comment]
I think that's it. When I migrated from HBP v1 to v2, I updated the Hasura container to this:

      HASURA_GRAPHQL_JWT_SECRET: '{"type": "RS256", "jwk_url": "http://auth:3000/auth/jwks"}'
      
I also took the depricated JWT_SECRET_KEY out of HBP container, but I never added a JWT_KEY or JWT_KEY_FILE.

To try to resolve this I did the following:
ssh-keygen -t rsa -b 2048 -m PEM -f jwtRS256.key
openssl rsa -in jwtRS256.key -pubout -outform PEM -out jwtRS256.key.pub

And then added that to the docker-container.yaml:

```
  auth:
    image: nhost/hasura-backend-plus:latest
    environment:
...
      JWT_KEY_FILE_PATH:/app/jwtRS256.key
    volumes:
      - ./jwtRS256.key:/app/jwtRS256.key
      - ./jwtRS256.key.pub:/app/jwtRS256.key.pub

```
I have spent at least half an restarting the container with different configurations, but each time I get a different response from auth:3000/auth/jwks. What am I doing wrong?

**Comment by elitan** [comment]
I'll ping in @plmercereau who's the expert on JWKS. Hopefully he can help you.

**Comment by JKing-B16** [comment]
So I came back to this after a week and figured out the issue. The JWT_KEY_FILE_PATH environmental variable is not currently being used, but rather always falling back to a default value. I have a pull request to fix this:

 Use JWT_KEY_FILE_PATH if available #467 

In the mean time I also have the following workaround. In docker-compose.yaml, you can map your key to the default key location:
`
    volumes:
      - ./jwtRS256.key:/app/custom/keys/private.pem
`
